hieroplatform/src/main/java/org/hiero/sketch/table/RecordEq.java

```diff
@@ -0,0 +1,57 @@
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.table.api.ISubSchema;
+import org.hiero.sketch.table.api.ITable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordEq {
+    private final List<ColumnDescription> colNames;
+
+    public RecordEq() {
+        this.colNames = new ArrayList<ColumnDescription>();
+    }
+
+    public void append(ColumnDescription colDesc) {
+        this.colNames.add(colDesc);
+    }
+
+    /**
+     * Return a Schema describing the columns used to check Equality.
+     */
+    public Schema toSchema() {
```

Comment:
These methods are strange.
I wonder whether it wouldn't be better to have a Schema in this class.
Also, the toSubSchema() method looks like it could be in the Schema class.